### PR TITLE
Core: Add `shortcuts` URL param to disable keyboard shortcuts

### PR DESCRIPTION
**Pull request by ghengeveld**
Issue: #15172

## What I did

This adds a new `shortcuts` query param for the manager. When set to `false` it will set `ui.enableShortcuts: false`, so that keyboard shortcuts will be disabled. This is particularly useful when embedding stories on a page, where keyboard shortcuts might conflict.

Additionally, I've taken the opportunity to support `true` and `false` as alternative to `0` or `1`, add a table of supported params to the docs, and properly deprecate some legacy params.

## How to test

- Is this testable with Jest or Chromatic screenshots? no
- Does this need a new example in the kitchen sink apps? no
- Does this need an update to the documentation? yes, added

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

File: lib/api/src/modules/url.ts [hunk 1]
```diff
@@ -42,51 +35,69 @@ let prevParams: ReturnType<typeof queryFromLocation>;
...
+  }
+  if (shortcuts === 'false' || shortcuts === '0') {
+    ui.enableShortcuts = false;
+  }
+  if (addonPanel) {
+    selectedPanel = addonPanel;
   }
 
-  // Legacy URLs
+  // @deprecated Superceded by `panel=false`, to be removed in 7.0
```
Comment (shilman) [inline lib/api/src/modules/url.ts:78]: Can you add this to MIGRATION.md as well?

**Review by shilman — CHANGES_REQUESTED** [review]
Very nice. Love the code improvements too! One minor request.

File: lib/api/src/modules/url.ts (hunk 1 above)
Comment (ghengeveld) [reply lib/api/src/modules/url.ts:78]: Done

**Comment by ghengeveld** [comment]
`e2e-tests-pnp` was already broken on `next`.

**Review by shilman — APPROVED** [review]
🚀

**Pull request merged**